agent/eni/netwrapper/netwrapper_windows.go

```diff
@@ -0,0 +1,88 @@
+//go:build windows
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package netwrapper
+
+import (
+	"encoding/json"
+	"net"
+	"os/exec"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+type WindowsNetworkAdapter struct {
+	Name       string `json:"ifAlias"`
+	MacAddress string `json:"MacAddress"`
+	IfIndex    int    `json:"ifIndex"`
+	MTU        int    `json:"ActiveMaximumTransmissionUnit"`
+}
+
+type WinAdapters []WindowsNetworkAdapter
+
+// Making it visible for unit testing
+var execCommand = exec.Command
+
+// getAllNetworkInterfaces returns all the network adapters.
+// Unfortunately, as of August 2022, the Win32 IPHelper API seems to be flaky wherein
+// the same stops responding in few intermittent cases. Therefore, to avoid failures
+// in such cases, we have switched to using Powershell cmdlet for fetching all the interfaces.
+func (utils *utils) getAllNetworkInterfaces() ([]net.Interface, error) {
+	log.Info("Running Powershell command to fetch all the network adapters")
+
+	// This command returns a JSON array of all the network interfaces on the instance.
+	cmd := "Get-NetAdapter | select ifAlias, ifIndex, MacAddress, ActiveMaximumTransmissionUnit | ConvertTo-Json"
+	out, err := execCommand("Powershell", "-C", cmd).Output()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to execute Get-NetAdapter for retrieving all interfaces")
+	}
+	log.Debugf("Output of Get-NetAdapters Powershell command execution: %s", out)
+
+	// Convert the obtained JSON array into an array of structs.
+	data := WinAdapters{}
```

Comment:
I'm pretty sure you can `var data WinAdapters` then `&data` below. You dont need to initialize to 0 size.